src/protos/io/defang/v1/fabric.proto

```diff
@@ -81,6 +81,11 @@ service FabricController {
     option idempotency_level = NO_SIDE_EFFECTS;
   }; // no values
 
+  rpc PutDeployment(Deployment) returns (google.protobuf.Empty);
```

Comment:
Should this be idempotent? (I think so)

Reply:
Ah yes. I'm not sure how exactly this idempotency signal affects actual client behaviour though. I should note that a timestamp is part of the request body. So if that timestamp changes, another item will be inserted into the backedn store.

```suggestion
  rpc PutDeployment(Deployment) returns (google.protobuf.Empty) {
    option idempotency_level = IDEMPOTENT;
  }
```

Reply:
I think the main difference it makes for the client library is its retry behaviour, it could be retried multiple times without worrying having an inaccurate result, and in this case, the timestamp being updated to the slightly later retry time should be considered okay for idempotency.

Reply:
So in this case, the client will re-initiate the request with the original parameters, right?